lib/fragment.js

```diff
@@ -80,33 +83,53 @@ module.exports = class Fragment extends EventEmitter {
      * Handles fetching the fragment
      * @param {object} request - HTTP request stream
      * @param {boolean} isFallback - decides between fragment and fallback URL
-     * @returns {object} Fragment response streams in case of synchronous fragment or buffer incase of async fragment
+     * @param {object} parentSpan - opentracing Span that will be the parent of the current operation
+     * @returns {object} Fragment response streams in case of synchronous fragment or buffer in case of async fragment
      */
-    fetch(request, isFallback) {
+    fetch(request, isFallback, parentSpan = null) {
         if (!isFallback) {
             this.emit('start');
         }
         const url = isFallback
             ? this.attributes.fallbackUrl
             : this.attributes.url;
 
-        this.requestFragment(url, this.attributes, request).then(
-            res => this.onResponse(res, isFallback),
-            err => {
-                if (!isFallback) {
-                    const { fallbackUrl } = this.attributes;
-                    if (fallbackUrl) {
-                        this.emit('fallback', err);
-                        this.fetch(request, true);
+        const spanOptions = parentSpan ? { childOf: parentSpan } : {};
+        const span = tracer.startSpan('fetch_fragment', spanOptions);
+        span.setTag('fragment-url', url);
+        span.setTag('fragment-id', this.attributes.id || 'unnamed');
+        span.setTag('fallback', isFallback);
+        span.setTag('primary', this.attributes.primary || false);
```

Comment:
we dont need || false here, we already assign default values in the constructor 
https://github.com/zalando/tailor/blob/477c65166f2c09e3c6f5928ff87b4047b5dc6215/lib/fragment.js#L61-L69

Reply:
Cool. Will remove.